builtin/providers/aws/resource_aws_network_interface.go

```diff
@@ -0,0 +1,307 @@
+package aws
+
+import (
+	"bytes"	
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+	
+	"github.com/hashicorp/aws-sdk-go/aws"
+	"github.com/hashicorp/terraform/helper/hashcode"	
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/aws-sdk-go/gen/ec2"
+)
+
+func resourceAwsNetworkInterface() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNetworkInterfaceCreate,
+		Read:   resourceAwsNetworkInterfaceRead,
+		Update: resourceAwsNetworkInterfaceUpdate,
+		Delete: resourceAwsNetworkInterfaceDelete,
+
+		Schema: map[string]*schema.Schema{			
+			
+			"subnet_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,				
+				ForceNew: true,
+			},
+
+			"private_ips": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,				
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+
+			"security_groups": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,				
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+
+			"attachment": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"instance": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"device_index": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"attachment_id": &schema.Schema{
+							Type:     schema.TypeString,	
+							Computed: true,						
+						},
+					},
+				},
+				Set: resourceAwsEniAttachmentHash,
+			},
+
+			"tags": tagsSchema(),			
+		},
+	}
+}
+
+func resourceAwsNetworkInterfaceCreate(d *schema.ResourceData, meta interface{}) error {
+	
+	ec2conn := meta.(*AWSClient).awsEC2conn
+
+	request := &ec2.CreateNetworkInterfaceRequest{		
+		Groups: 				expandStringList(d.Get("security_groups").(*schema.Set).List()),
+		SubnetID:				aws.String(d.Get("subnet_id").(string)),
+		PrivateIPAddresses:		convertToPrivateIPAddresses(d.Get("private_ips").(*schema.Set).List()),
+	}
+	
+	log.Printf("[DEBUG] Creating network interface")
+	resp, err := ec2conn.CreateNetworkInterface(request)
+	if err != nil {
+		return fmt.Errorf("Error creating ENI: %s", err)
+	}
+
+	new_interface_id := *resp.NetworkInterface.NetworkInterfaceID
+	d.SetId(new_interface_id)
```

Comment:
small nitpick in that `new_interface_id` is never used beyond setting here, so `d.SetId(*resp.NetworkInterface.NetworkInterfaceID)` works just as well. Don't feel you _need_ to change this though :smile: